powershell-yaml.psd1

```diff
@@ -25,7 +25,7 @@
 RootModule = 'powershell-yaml.psm1'
 
 # Version number of this module.
-ModuleVersion = '0.3.7'
+ModuleVersion = '0.3.8'
```

Comment:
Considering this upgrades ```YamlDotNet```, I think you can bump it to ```0.4.0```

Reply:
Given this, do you want the change to empty node parsing as $null instead of "" to be behind a switch on ConvertFrom-Yaml, or should I just make the change?

Reply:
Tell you what.  It's getting late here so I'll push with the null parsing change included as default behaviour, and go back and add switches tomorrow if you'd prefer it implemented that way.

Reply:
You can make the change. I am ok even if you decide to keep the ```if``` statements as they are. Possibly just make a function out of them, just to make it cleaner. My comment was more of a suggestion than a demand.

In regards to empty nodes being quoted as empty double quotes, this is actually desirable. In some cases you want to set a field as an empty string, and you want to be able to properly deserialize it, even on different languages.

So for example here is how python yaml module does it:

```python
>>> import yaml
>>> a = {"a": "", "b":None}
>>> print(yaml.dump(a))
{a: '', b: null}

>>> 
>>> q = """
... ---
... a:
... b: ""
... """
>>> 
>>> ww = yaml.load(q)
>>> ww
{'a': None, 'b': ''}
```
So to conclude, an empty string should be serialized as ```""```. A ```$null``` should be an empty scalar, or an explicit ```null```.

Reply:
Yep, that's what is now implemented.  The change in `Convert-ValueToProperType` is to work around YamlDotNet failing to _parse_ empty scalars as null - with that in place, powershell-yaml should now match both of those pyyaml examples, instead of producing `@{a="";b=""}` on load :)

Reply:
Cool. So this test case should work:

```powershell
PS /tmp/aaa/powershell-yaml> $nullAndString = @{"iAmNull"= $null; "iAmEmptyString"=""}
PS /tmp/aaa/powershell-yaml> $backToYaml = ConvertFrom-Yaml $asYaml                   
PS /tmp/aaa/powershell-yaml> $asYaml = ConvertTo-Yaml $nullAndString
PS /tmp/aaa/powershell-yaml> $asYaml
iAmEmptyString: ""
iAmNull: 

PS /tmp/aaa/powershell-yaml> $backToYaml = ConvertFrom-Yaml $asYaml
PS /tmp/aaa/powershell-yaml> $nullAndString["iAmNull"] -eq $backToYaml["iAmNull"]
True              
PS /tmp/aaa/powershell-yaml> $nullAndString["iAmString"] -eq $backToYaml["iAmString"]
True
```
If yes, then awesome :).

Reply:
Lemme add that as an actual test case

Reply:
Done. :)